prep_TS_hyp3.ipynb

```diff
@@ -0,0 +1,500 @@
+    "        outfiles.append(outfile)\n",
+    "\n",
+    "    return(outfiles)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Clip the files"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# get files to be clipped in the outdir and its sub-directories.\n",
+    "files = []\n",
+    "pattern = [\"*.tif\"]                      \n",
+    "for dir,_,_ in os.walk(outdir):\n",
+    "    for item in pattern:\n",
+    "        files.extend(glob.glob(os.path.join(dir, item)))\n",
+    "# clip off files\n",
+    "cutGeotiffsByLine(files)    "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Copy metadata file and dem file to DEM subdirectory"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# copy the *.txt and dem.tif in one subdirctory to DEM subdirectory\n",
+    "cmd = \"mkdir -p {}/DEM {}/inputs\".format(outdir, outdir)\n",
+    "! {cmd}\n",
+    "prefix = glob.glob('./*.zip')[0].split(\".zip\")[0]\n",
+    "fileprefix = os.path.basename(prefix)\n",
+    "txtfile = \"{dir}/{fileprefix}.txt\".format(dir = prefix, fileprefix = fileprefix)\n",
+    "cmd = \"cp {file} {data_dir}/DEM/dem_clip.txt\".format(file = txtfile, data_dir = outdir)\n",
+    "! {cmd}\n",
+    "demfile = \"{dir}/{fileprefix}_dem_clip.tif\".format(dir = prefix, fileprefix = fileprefix)\n",
+    "cmd = \"cp {file} {data_dir}/DEM/dem_clip.tif\".format(file = demfile, data_dir = outdir)    \n",
+    "! {cmd}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(\"prepared data are at {}\".format(outdir))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Prepared data structure are:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "! tree {outdir}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Configure the custom cfg file in the outdir"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Show the cunstom cfg file  "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from pathlib import Path\n",
+    "\n",
+    "config = f'''\n",
+    "mintpy.load.processor        = hyp3\n",
+    "##---------interferogram datasets:\n",
+    "mintpy.load.unwFile          = {outdir}/*/*unw_phase_clip.tif\n",
+    "mintpy.load.corFile          = {outdir}/*/*corr_clip.tif\n",
+    "##---------geometry datasets:\n",
+    "mintpy.load.demFile          = {outdir}/DEM/dem_clip.tif\n",
+    "##---------incAngleFile:\n",
+    "mintpy.load.incAngleFile    =  {outdir}/*/*inc_map_clip.tif\n",
```

Comment:
Testing suggests we can use wildcards for both the dem and the inc_map, and *one* of the matching files will be chosen and used for processing.  That should be quick and easy for most use cases, and should eliminate the need for the earlier `Copy metadata file and dem file to DEM subdirectory`
```suggestion
    "##---------geometry datasets:\n",
    "mintpy.load.demFile          = {outdir}/*/*dem_clip.tif\n",
    "mintpy.load.incAngleFile     =  {outdir}/*/*inc_map_clip.tif\n",
```

Reply:
I suggest keep as it is. The smallbaselineApp loads the dem_clip.tif and the metadata file which is called dem_clip.txt.  If follow your suggestion, we have to rename every *.txt file into dem_clip.txt.  I think current config file make a clear idea how we get the dem.tif and its metadata.txt.

Reply:
I second @asjohnston-asf suggestion. With the latest version from Forrest, it works from my testing. Plus, no need for manual DEM dir.

Reply:
changed as your suggestion